### ISSUE TITLE
why ap ap50 ap75 = 0.000?

### ISSUE THREAD

**Issue by oopsboy** [facebookresearch/adaptive_teacher#20]
Train

**Comment by wha-cloud** [comment]
> ![222](https://user-images.githubusercontent.com/63640150/175491267-cc7c84c1-dc78-4a1c-b1df-600a9bd5ec9b.png)

I have the same problem as you, did you solve it? and my total loss became nan

**Comment by yujheli** [comment]
@wha-cloud Could you check if the learning rate is too large? Or if you use too small batch size? See: https://github.com/facebookresearch/adaptive_teacher/issues/9

**Comment by wha-cloud** [comment]
> @wha-cloud Could you check if the learning rate is too large? Or if you use too small batch size? See: #9

Thanks! When I set the batch size to 8, up to now 2000 iterations are no problem. If it is set to 4, there are problems with several hundred iterations. Now I have a new question, it seems to be evaluated twice when training, why is the result of the second evaluation 0?
<img width="944" alt="image" src="https://user-images.githubusercontent.com/65757396/176449166-f8516ebd-fe56-4364-8855-494f4facc855.png">

**Comment by oopsboy** [comment]
> 

I wonder if you have solved it?

**Comment by yujheli** [comment]
@wha-cloud @oopsboy Does the number in tensorboard make sense? I always looked at the number in tensorboard and rarely see the numbers printed out.

**Comment by helq2612** [comment]
@[wha-cloud](https://github.com/wha-cloud) I guess you are in the source only pre-training stage, in which the student model does not have learned weights yet (https://github.com/facebookresearch/adaptive_teacher/blob/cba3c59cadfc9f1a3a676a82bf63d76579ab552b/configs/faster_rcnn_R101_cross_clipart.yaml#L43).

**Comment by oopsboy** [comment]
> @[wha-cloud](https://github.com/wha-cloud) I guess you are in the source only pre-training stage, in which the student model does not have learned weights yet (
> 
> https://github.com/facebookresearch/adaptive_teacher/blob/cba3c59cadfc9f1a3a676a82bf63d76579ab552b/adapteacher/engine/trainer.py#L506
> 
> ).

@helq2612 Thank you for your answer. How can I modify the number of times of this paramete

**Comment by helq2612** [comment]
@[oopsboy](https://github.com/oopsboy) I think you can modify it in the config files, e.g.:
https://github.com/facebookresearch/adaptive_teacher/blob/cba3c59cadfc9f1a3a676a82bf63d76579ab552b/configs/faster_rcnn_R101_cross_clipart.yaml#L43
Or pass it as args to train_net.py

**Comment by yujheli** [comment]
Ok, the model will be evaluated two times: 
If you are in the stage of burn-in, you will get 0 AP for teacher.
bbox (not uses):
<img width="1110" alt="image" src="https://user-images.githubusercontent.com/23270045/178135027-7fefca3f-4aa4-47f4-846b-0cb40655de82.png">
bbox_student:
<img width="1157" alt="image" src="https://user-images.githubusercontent.com/23270045/178135055-a958db4b-f646-4f30-b6bd-172021f9e7b8.png">
